packages/runner-ct/src/plugins/UIPlugin.ts

```diff
@@ -1,7 +1,8 @@
 export type UIPlugin = {
   name: string
   type: 'devtools'
-  initialize: (contentWindow: Window) => void
-  mount: () => void
-  unmount: () => void
+  initialize?: (contentWindow: Window) => void
+  mount?: () => void
+  unmount?: () => void
+  beforeTest?: () => void
```

Comment:
I`d say that we can rename this to `onBeforeTest` to be more declarative, but it is good to go we will need to rethink the architecture anyway